persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -64,7 +64,12 @@ open' connStr logFunc = do
             | Just cs <- T.stripPrefix "WAL=off " connStr -> (cs, False)
             | otherwise                                   -> (connStr, True)
 
-    conn <- Sqlite.open connStr'
+    let (connStr'', readOnly) = case () of
+          ()
+            | Just cs <- T.stripPrefix "READONLY " connStr'  -> (cs, True)
+            | Just cs <- T.stripPrefix "READWRITE " connStr' -> (cs, False)
+            | otherwise                                      -> (connStr', False)
+    conn <- Sqlite.open connStr'' readOnly
```

Comment:
If I'm reading this code right, does this mean you have to have `WAL=on READONLY databaseName` in exactly that  order (i.e. you can't do `READONLY WAL=on databaseName`)? If so that seems like it'll mess people up